packages/ui5-proxy-middleware/src/base/ui5.d.ts

```diff
@@ -0,0 +1,118 @@
+declare module '@ui5/fs' {
+    /**
+     * https://sap.github.io/ui5-tooling/api/module-@ui5_fs.Resource.html
+     */
+    export class Resource {
+        /**
+         * Gets the resources path
+         */
+        getPath(): string;
+
+        /**
+         * Gets a buffer with the resource content.
+         */
+        getBuffer(): Promise<Buffer>;
+
+        /**
+         * Gets a string with the resource content.
+         */
+        getString(): Promise<string>;
+    }
+
+    /**
+     * https://sap.github.io/ui5-tooling/stable/api/@ui5_fs_ReaderCollection.html
+     */
+    export class ReaderCollection {
+        /**
+         * Locates resources by matching glob patterns.
+         */
+        byGlob(virPattern: string | string[], options?: object): Promise<Resource[]>;
+
+        /**
+         * Locates resources by matching a given path.
+         */
+        byPath(virPattern: string | string[], options?: object): Promise<Resource[]>;
+    }
+
+    /**
+     * https://sap.github.io/ui5-tooling/stable/api/module-@ui5_fs.DuplexCollection.html
+     */
+    export class DuplexCollection extends ReaderCollection {}
+
+    /**
+     * https://sap.github.io/ui5-tooling/api/module-@ui5_fs.AbstractReader.html
+     */
+    export class AbstractReader {}
+}
+
+declare module '@ui5/server' {
+    export interface MiddlewareParameters<C> {
+        /**
+         * DuplexCollection to read and write files
+         */
+        resources: {
+            all: ReaderCollection;
+            dependencies: ReaderCollection;
+            rootProject: ReaderCollection;
+        };
+
+        /**
+         * Project specific options
+         */
+        options: {
+            /**
+             * Optional middleware configuration if provided in ui5*.yaml
+             */
+            configuration?: C;
+        };
+
+        /**
+         * Middleware utilities (not yet used in our middlewares)
+         */
+        middlewareUtils: unknown;
+    }
+}
+
+declare module '@ui5/builder' {
+    /**
+     * https://sap.github.io/ui5-tooling/api/module-@ui5_builder.tasks.TaskUtil.html
```

Comment:
```suggestion
     * https://sap.github.io/ui5-tooling/v3/api/@ui5_project_build_helpers_TaskUtil.html
```
Broken link. v3 replacement link